gui/kitchen-info.lua

```diff
@@ -0,0 +1,433 @@
+                    drawNumberColumn(numberStart, kitchen, firstVisibleIndex, numDisplayedItems)
+                end
+
+                drawPermissionsColumn(permissionsStart, kitchen, firstVisibleIndex, numDisplayedItems)
+            end
+
+            -- Column labels
+            local labels_p = gui.Painter.new_xy(columnsPos, 4, gps.dimx - 2, 4)
+            labels_p:pen(COLOR_WHITE)
+
+            for i, column in ipairs(visibleColumns) do
+                labels_p:seek(columnOffset[i], 0):string(column.label)
+            end
+
+            -- Columns data
+            local p = gui.Painter.new_xy(columnsPos, 6, gps.dimx - 3, 6 + numDisplayedItems - 1)
+
+            local item_type    = kitchen.item_type[kitchen.page]
+            local item_subtype = kitchen.item_subtype[kitchen.page]
+            local mat_type     = kitchen.mat_type[kitchen.page]
+            local mat_index    = kitchen.mat_index[kitchen.page]
+
+            for i = 0, numDisplayedItems - 1 do
+                local index = firstVisibleIndex + i
+                local matinfo = dfhack.matinfo.decode(mat_type[index], mat_index[index])
+                local material = matinfo.material
+                local plant_raw = df.plant_raw.find( mat_index[index] )
+
+                for c, column in ipairs(visibleColumns) do
+                    local columnValue = column.getColumnValue(item_type[index], material, plant_raw)
+
+                    p:seek(columnOffset[c], i)
+                    if (columnValue) then
+                        p:advance(column.minValueLeftPadding)
+                        p:pen(COLOR_LIGHTGREEN):string(columnValue)
+                    else
+                        p:advance(column.emptyValueLeftPadding)
+                        p:pen(COLOR_DARKGREY):string("----")
+                    end
+                end
+            end
+        end
+    end
+
+    -- Cursor
+    local cursorY = kitchen.cursor - firstVisibleIndex
+
+    if (numDisplayedItems >= 1 and cursorY < numDisplayedItems) then
+        local cursor_p = gui.Painter.new_xy(0, 6 + cursorY, gps.dimx - 1, 6 + cursorY)
+
+        cursor_p:pen(COLOR_LIGHTGREEN)
+        --cursor_p:seek(1, 0):char('>')
+        cursor_p:seek(permissionsStart, 0):char('[')
+        cursor_p:seek(permissionsEnd, 0):char(']')
+        if show_processing then
+            cursor_p:seek(gps.dimx - 2, 0):char('<')
+        end
+    end
+
+    -- Hotkey
+    local keys_p = gui.Painter.new_xy(1, gps.dimy - 3, gps.dimx - 2, gps.dimy - 2)
+    local keys_x = math.min(gps.dimx - 3 - string.len("p: Hide processing info"), 62)
+    keys_p:key_pen(COLOR_LIGHTRED):pen(COLOR_WHITE)
+    keys_p:seek(keys_x, 1):key('CUSTOM_P'):string(show_processing and ": Hide processing info" or ": Show processing info")
+end
+
+function kitchen_overlay:onInput(keys)
+    if keys.CUSTOM_P then
+        show_processing = not show_processing
```

Comment:
this does not play nice with the search filter. Every time a 'p' appears in the search box, the gui/kitchen-info info is toggled.

I actually think we can get away with not having this option at all. What does toggling add? If a player wants it gone, they can just run `disable gui/kitchen-info`